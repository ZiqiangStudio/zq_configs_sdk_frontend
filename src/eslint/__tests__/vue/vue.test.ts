import { exec, ExecException } from 'child_process';
import { promisify } from 'util';
import { fileURLToPath } from 'url';
import { dirname } from 'path';

const isError = (e: unknown): e is Error => {
  return (
    typeof e === 'object' &&
    !!e &&
    'name' in e &&
    typeof e.name === 'string' &&
    'message' in e &&
    typeof e.message === 'string'
  );
};

const isExecException = (e: unknown): e is ExecException => {
  return isError(e) && 'cmd' in e && typeof e.cmd === 'string';
};

const execEslint = async ({ file }: { file: string }) => {
  const promiseExec = promisify(exec);
  const dirName = dirname(fileURLToPath(import.meta.url));
  try {
    const { stdout } = await promiseExec(
      `npx eslint -c ${dirName}/eslint.config.js ${dirName}/${file}`,
    );
    return stdout;
  } catch (e) {
    if (isExecException(e)) {
      return e.stdout ?? '';
    }
    return '';
  }
};

describe('Vue Lint', () => {
  test('Vue Lint 报错', async () => {
    const eslintOutput = await execEslint({ file: 'vue.vue' });
    expect(eslintOutput).toMatchSnapshot();
  });
});
