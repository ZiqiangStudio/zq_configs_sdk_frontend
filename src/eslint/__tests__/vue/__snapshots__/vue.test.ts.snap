// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test test2 1`] = `
"
/Users/bytedance/dev/zq_configs_sdk_frontend/src/eslint/__tests__/vue/vue.vue
   2:1   error  Insert \`··\`                                                                prettier/prettier
   3:1   error  Replace \`··<p·class="title">{title}·\` with \`····<p·class="title">{title}\`  prettier/prettier
   4:1   error  Insert \`··\`                                                                prettier/prettier
   7:1   error  Delete \`··\`                                                                prettier/prettier
   7:3   error  'onMount' is not defined                                                   no-undef
  13:1   error  Delete \`⏎\`                                                                 prettier/prettier
  17:18  error  Insert \`;\`                                                                 prettier/prettier

✖ 7 problems (7 errors, 0 warnings)
  6 errors and 0 warnings potentially fixable with the \`--fix\` option.

"
`;
